### ISSUE TITLE
🐛 | Does not recognize USB RFID-Reader in Web UI

### ISSUE THREAD

**Issue by jonasschmid404** [MiczFlor/RPi-Jukebox-RFID#2457]
### Version

2.8.0 - 5044358f

### Branch

master

### OS

Raspbian GNU/Linux 11 (bullseye)

### Pi model

3 B v1.2

### Hardware

[USB RFID-Reader (Neuftech for EM4100)](https://www.amazon.de/gp/product/B018OYOR3E?&linkCode=sl1&tag=diebarnfeld-21&linkId=f924b3d7972dadb096709b95f757ca72&language=de_DE&ref_=as_li_ss_tl)

### What happened?

Hi everyone, so I'm pretty new to this thing - but very enthusiastic about this project. So I have a problem: when swiping an RFID-card, it is read in the command line interface ("-bash: 00055774785: command not found"), but not in the web UI (javascript is activated. I also can't enter anything manually in the corresponding field "Last Used Chip ID". The issue is somewhat related to issue #768, but not quite.  Thank you for your help, it is greatly appreciated!

**Comment by s-martin** [comment]
What’s the output of `tail -n 200 /var/log/syslog`?

**Comment by jonasschmid404** [comment]
sorry, I'm too stupid. How can I copy the log from the command line in rasperry?

**Comment by s-martin** [comment]
Depends what ssh client you use, but usually you can select the output with your mouse and select all with `CTRL+A` and then copy.

**Comment by jonasschmid404** [comment]
I don't have an ssh client set up, I'm watching the raspberry output locally... Ok, so I'll do this first.

**Comment by jonasschmid404** [comment]
Here it is...

```
 Nov 25 22:42:08 PhonieboxGPIOControl[27330]:     device = self.generate_device(config[section], section)
Nov 25 22:42:08 audiobox PhonieboxGPIOControl[27330]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:08 audiobox PhonieboxGPIOControl[27330]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:08 audiobox PhonieboxGPIOControl[27330]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:08 audiobox PhonieboxGPIOControl[27330]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:08 audiobox PhonieboxGPIOControl[27330]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:08 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:08 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
Nov 25 22:42:12 audiobox systemd[1]: phoniebox-gpio-control.service: Scheduled restart job, restart counter is at 617.
Nov 25 22:42:12 audiobox systemd[1]: Stopped Phoniebox GPIO Control Service.
Nov 25 22:42:12 audiobox systemd[1]: Started Phoniebox GPIO Control Service.
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:GPIO Started
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:['VolumeControl', 'RotaryVolumeControl', 'PrevNextControl', 'PlayPause', 'Shutdown', 'PauseShutdown', 'Volume0', 'VolumeUp', 'VolumeDown', 'NextSong', 'PrevSong', 'FastForward', 'Rewind', 'Halt', 'RFIDDevice', 'TriggerPlayCard1', 'TriggerPlayFolderSomeRelativeFolderName', 'StatusLED']
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:Device VolumeControl not enabled
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:adding GPIO-Device, RotaryVolumeControl
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:adding GPIO-Device, PrevNextControl
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: /home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py:90: RuntimeWarning: A physical pull up resistor is fitted on this channel!
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:   GPIO.setup(self.pin, GPIO.IN, pull_up_down=self.pull_up_down)
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:GPIODevices.two_button_control:adding new action
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:GPIODevices.simple_button:PrevNextControlBtn1: set when_pressed
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:GPIODevices.simple_button:PrevNextControlBtn2: set when_pressed
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:Device PlayPause not enabled
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: INFO:__main__:adding GPIO-Device, Shutdown
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: RotaryVolumeControl
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: PrevNextControl
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: Shutdown
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: Traceback (most recent call last):
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 140, in <module>
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:     devices = gpio_control_class.get_all_devices(config)
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 108, in get_all_devices
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:     device = self.generate_device(config[section], section)
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:12 audiobox PhonieboxGPIOControl[27371]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:12 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:12 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
Nov 25 22:42:16 audiobox systemd[1]: phoniebox-gpio-control.service: Scheduled restart job, restart counter is at 618.
Nov 25 22:42:16 audiobox systemd[1]: Stopped Phoniebox GPIO Control Service.
Nov 25 22:42:16 audiobox systemd[1]: Started Phoniebox GPIO Control Service.
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:GPIO Started
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:['VolumeControl', 'RotaryVolumeControl', 'PrevNextControl', 'PlayPause', 'Shutdown', 'PauseShutdown', 'Volume0', 'VolumeUp', 'VolumeDown', 'NextSong', 'PrevSong', 'FastForward', 'Rewind', 'Halt', 'RFIDDevice', 'TriggerPlayCard1', 'TriggerPlayFolderSomeRelativeFolderName', 'StatusLED']
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:Device VolumeControl not enabled
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:adding GPIO-Device, RotaryVolumeControl
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:adding GPIO-Device, PrevNextControl
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: /home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py:90: RuntimeWarning: A physical pull up resistor is fitted on this channel!
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:   GPIO.setup(self.pin, GPIO.IN, pull_up_down=self.pull_up_down)
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:GPIODevices.two_button_control:adding new action
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:GPIODevices.simple_button:PrevNextControlBtn1: set when_pressed
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:GPIODevices.simple_button:PrevNextControlBtn2: set when_pressed
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:Device PlayPause not enabled
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: INFO:__main__:adding GPIO-Device, Shutdown
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: RotaryVolumeControl
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: PrevNextControl
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: Shutdown
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: Traceback (most recent call last):
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 140, in <module>
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:     devices = gpio_control_class.get_all_devices(config)
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 108, in get_all_devices
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:     device = self.generate_device(config[section], section)
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:16 audiobox PhonieboxGPIOControl[27411]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:16 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:16 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
Nov 25 22:42:20 audiobox systemd[1]: phoniebox-gpio-control.service: Scheduled restart job, restart counter is at 619.
Nov 25 22:42:20 audiobox systemd[1]: Stopped Phoniebox GPIO Control Service.
Nov 25 22:42:20 audiobox systemd[1]: Started Phoniebox GPIO Control Service.
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:GPIO Started
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:['VolumeControl', 'RotaryVolumeControl', 'PrevNextControl', 'PlayPause', 'Shutdown', 'PauseShutdown', 'Volume0', 'VolumeUp', 'VolumeDown', 'NextSong', 'PrevSong', 'FastForward', 'Rewind', 'Halt', 'RFIDDevice', 'TriggerPlayCard1', 'TriggerPlayFolderSomeRelativeFolderName', 'StatusLED']
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:Device VolumeControl not enabled
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:adding GPIO-Device, RotaryVolumeControl
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:adding GPIO-Device, PrevNextControl
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: /home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py:90: RuntimeWarning: A physical pull up resistor is fitted on this channel!
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:   GPIO.setup(self.pin, GPIO.IN, pull_up_down=self.pull_up_down)
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:GPIODevices.two_button_control:adding new action
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:GPIODevices.simple_button:PrevNextControlBtn1: set when_pressed
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:GPIODevices.simple_button:PrevNextControlBtn2: set when_pressed
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:Device PlayPause not enabled
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: INFO:__main__:adding GPIO-Device, Shutdown
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: RotaryVolumeControl
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: PrevNextControl
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: Shutdown
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: Traceback (most recent call last):
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 140, in <module>
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:     devices = gpio_control_class.get_all_devices(config)
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 108, in get_all_devices
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:     device = self.generate_device(config[section], section)
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:20 audiobox PhonieboxGPIOControl[27452]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:20 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:21 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
Nov 25 22:42:24 audiobox systemd[1]: phoniebox-gpio-control.service: Scheduled restart job, restart counter is at 620.
Nov 25 22:42:24 audiobox systemd[1]: Stopped Phoniebox GPIO Control Service.
Nov 25 22:42:24 audiobox systemd[1]: Started Phoniebox GPIO Control Service.
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:GPIO Started
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:['VolumeControl', 'RotaryVolumeControl', 'PrevNextControl', 'PlayPause', 'Shutdown', 'PauseShutdown', 'Volume0', 'VolumeUp', 'VolumeDown', 'NextSong', 'PrevSong', 'FastForward', 'Rewind', 'Halt', 'RFIDDevice', 'TriggerPlayCard1', 'TriggerPlayFolderSomeRelativeFolderName', 'StatusLED']
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:Device VolumeControl not enabled
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:adding GPIO-Device, RotaryVolumeControl
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:adding GPIO-Device, PrevNextControl
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: /home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py:90: RuntimeWarning: A physical pull up resistor is fitted on this channel!
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:   GPIO.setup(self.pin, GPIO.IN, pull_up_down=self.pull_up_down)
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:GPIODevices.two_button_control:adding new action
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:GPIODevices.simple_button:PrevNextControlBtn1: set when_pressed
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:GPIODevices.simple_button:PrevNextControlBtn2: set when_pressed
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:Device PlayPause not enabled
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: INFO:__main__:adding GPIO-Device, Shutdown
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: RotaryVolumeControl
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: PrevNextControl
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: Shutdown
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: Traceback (most recent call last):
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 140, in <module>
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:     devices = gpio_control_class.get_all_devices(config)
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 108, in get_all_devices
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:     device = self.generate_device(config[section], section)
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:24 audiobox PhonieboxGPIOControl[27493]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:24 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:24 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
Nov 25 22:42:28 audiobox systemd[1]: phoniebox-gpio-control.service: Scheduled restart job, restart counter is at 621.
Nov 25 22:42:28 audiobox systemd[1]: Stopped Phoniebox GPIO Control Service.
Nov 25 22:42:28 audiobox systemd[1]: Started Phoniebox GPIO Control Service.
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:GPIO Started
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:['VolumeControl', 'RotaryVolumeControl', 'PrevNextControl', 'PlayPause', 'Shutdown', 'PauseShutdown', 'Volume0', 'VolumeUp', 'VolumeDown', 'NextSong', 'PrevSong', 'FastForward', 'Rewind', 'Halt', 'RFIDDevice', 'TriggerPlayCard1', 'TriggerPlayFolderSomeRelativeFolderName', 'StatusLED']
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:Device VolumeControl not enabled
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:adding GPIO-Device, RotaryVolumeControl
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:adding GPIO-Device, PrevNextControl
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: /home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py:90: RuntimeWarning: A physical pull up resistor is fitted on this channel!
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:   GPIO.setup(self.pin, GPIO.IN, pull_up_down=self.pull_up_down)
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:GPIODevices.two_button_control:adding new action
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:GPIODevices.simple_button:PrevNextControlBtn1: set when_pressed
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:GPIODevices.simple_button:PrevNextControlBtn2: set when_pressed
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:Device PlayPause not enabled
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: INFO:__main__:adding GPIO-Device, Shutdown
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: RotaryVolumeControl
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: PrevNextControl
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: Shutdown
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: Traceback (most recent call last):
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 140, in <module>
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:     devices = gpio_control_class.get_all_devices(config)
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 108, in get_all_devices
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:     device = self.generate_device(config[section], section)
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:28 audiobox PhonieboxGPIOControl[27534]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:28 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:28 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
Nov 25 22:42:32 audiobox systemd[1]: phoniebox-gpio-control.service: Scheduled restart job, restart counter is at 622.
Nov 25 22:42:32 audiobox systemd[1]: Stopped Phoniebox GPIO Control Service.
Nov 25 22:42:32 audiobox systemd[1]: Started Phoniebox GPIO Control Service.
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:GPIO Started
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:['VolumeControl', 'RotaryVolumeControl', 'PrevNextControl', 'PlayPause', 'Shutdown', 'PauseShutdown', 'Volume0', 'VolumeUp', 'VolumeDown', 'NextSong', 'PrevSong', 'FastForward', 'Rewind', 'Halt', 'RFIDDevice', 'TriggerPlayCard1', 'TriggerPlayFolderSomeRelativeFolderName', 'StatusLED']
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:Device VolumeControl not enabled
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:adding GPIO-Device, RotaryVolumeControl
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:adding GPIO-Device, PrevNextControl
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: /home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py:90: RuntimeWarning: A physical pull up resistor is fitted on this channel!
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:   GPIO.setup(self.pin, GPIO.IN, pull_up_down=self.pull_up_down)
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:GPIODevices.two_button_control:adding new action
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:GPIODevices.simple_button:PrevNextControlBtn1: set when_pressed
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:GPIODevices.simple_button:PrevNextControlBtn2: set when_pressed
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:GPIODevices.simple_button:add new action
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:Device PlayPause not enabled
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: INFO:__main__:adding GPIO-Device, Shutdown
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: RotaryVolumeControl
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: PrevNextControl
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: Shutdown
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: Traceback (most recent call last):
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 140, in <module>
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:     devices = gpio_control_class.get_all_devices(config)
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 108, in get_all_devices
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:     device = self.generate_device(config[section], section)
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/gpio_control.py", line 62, in generate_device
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:     return SimpleButton(config.getint('Pin'),
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:   File "/home/pi/RPi-Jukebox-RFID/components/gpio_control/GPIODevices/simple_button.py", line 93, in __init__
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]:     GPIO.add_event_detect(self.pin, edge=self.edge, callback=self.callbackFunctionHandler,
Nov 25 22:42:32 audiobox PhonieboxGPIOControl[27575]: RuntimeError: Conflicting edge detection already enabled for this GPIO channel
Nov 25 22:42:32 audiobox systemd[1]: phoniebox-gpio-control.service: Main process exited, code=exited, status=1/FAILURE
Nov 25 22:42:32 audiobox systemd[1]: phoniebox-gpio-control.service: Failed with result 'exit-code'.
```

**Comment by s-martin** [comment]
Did you try to swipe a card right before you got the log file ?

**Comment by s-martin** [comment]
Probably unrelated, something’s wrong with your GPIO settings.
Can you post the content of `~/RPi-Jukebox-RFID/settings/gpio_settings.ini`?

**Comment by jonasschmid404** [comment]
> Did you try to swipe a card right before you got the log file ?

Yes, I did, but now it seems to work. Don't know what I did, or whether two-three reboots solved the problem. Thank you anyway!!!

Regarding the gpio_settings, here's the content of the command:

/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 1: [DEFAULT]: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 2: enabled:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 3: antibouncehack:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 5: [VolumeControl]: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 6: enabled:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 7: Type:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 8: Pin1:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 9: Pin2:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 10: pull_up_down:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 11: hold_time:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 12: hold_mode:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 13: functionCall1:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 14: functionCall2:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 15: functionCallTwoButtons:: command not found
/home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 16: syntax error near unexpected token ;' /home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini: line 16: ;functionCall1Args: 1'

I haven't yet connected the buttons, but I defined a shutdown button (Pin 3), a rotary volumne button (Pins 22 and 23), a Previous Song-Button (Pin 15) and a Next-Song-Button (Pin 16).

**Comment by s-martin** [comment]
Please use `cat ~/RPi-Jukebox-RFID/settings/gpio_settings.ini`

**Comment by jonasschmid404** [comment]
```
[DEFAULT]
enabled: True
antibouncehack: True

[VolumeControl]
enabled: False
Type: TwoButtonControl
Pin1: 5
Pin2: 6
pull_up_down: pull_up
hold_time: 0.3
hold_mode: Repeat
functionCall1: functionCallVolU
functionCall2: functionCallVolD
functionCallTwoButtons: functionCallVol0
;functionCall1Args: 1
;functionCall2Args: 1
;functionCallTwoButtonsArgs: x

[RotaryVolumeControl]
enabled: True
Type: RotaryEncoder
Pin1: 5
Pin2: 6
timeBase: 0.1
functionCall1: functionCallVolU
functionCall2: functionCallVolD

[PrevNextControl]
enabled: False
Type: TwoButtonControl
Pin1: 15
Pin2: 16
functionCall1: functionCallPlayerPrev
functionCall2: functionCallPlayerNext
functionCallTwoButtons: None
pull_up_down: pull_up
hold_time: 0.2
hold_mode: None

[PlayPause]
enabled: False
Type: Button
Pin: 27
pull_up_down: pull_up
functionCall: functionCallPlayerPause

[Shutdown]
enabled: True
Type:  Button
Pin: 3
pull_up_down: pull_up
hold_mode: Postpone
hold_time: 3.0
functionCall: functionCallShutdown

[PauseShutdown]
enabled: False
Type:  Button
Pin: 3
pull_up_down: pull_up
hold_time: 2.0
hold_mode: SecondFunc
functionCall: functionCallPlayerPause
functionCall2: functionCallShutdown

[Volume0]
enabled: False
Type:  Button
Pin: 17
pull_up_down: pull_up
functionCall: functionCallVol0

[VolumeUp]
enabled: False
Type:  Button
Pin: 16
pull_up_down: pull_up
hold_time: 0.3
hold_mode: Repeat
functionCall: functionCallVolU
;functionCallArgs: 1

[VolumeDown]
enabled: False
Type:  Button
Pin: 19
pull_up_down: pull_up
hold_time: 0.3
hold_mode: Repeat
functionCall: functionCallVolD
;functionCallArgs: 1

[NextSong]
enabled: True
Type:  Button
Pin: 16
pull_up_down: pull_up
functionCall: functionCallPlayerNext

[PrevSong]
enabled: True
Type:  Button
Pin: 15
pull_up_down: pull_up
functionCall: functionCallPlayerPrev

[FastForward]
enabled: False
Type:  Button
Pin: 7
pull_up_down: pull_up
functionCall: functionCallPlayerSeekFwd
;functionCallArgs: 10

[Rewind]
enabled: False
Type:  Button
Pin:
pull_up_down: pull_up
functionCall: functionCallPlayerSeekBack
;functionCallArgs: 10

[Halt]
enabled: False
Type:  Button
Pin: 25
pull_up_down: pull_up
functionCall: functionCallPlayerPauseForce

[RFIDDevice]
enabled: False
Type: Button
Pin: 21
pull_up_down: pull_up
functionCall: functionCallPlayerStop

[TriggerPlayCard1]
enabled: False
Type: Button
Pin: 4
pull_up_down: pull_up
functionCall: functionCallTriggerPlayCardId
functionCallArgs: 1

[TriggerPlayFolderSomeRelativeFolderName]
enabled: False
Type: Button
Pin: 4
pull_up_down: pull_up
functionCall: functionCallTriggerPlayFolder
functionCallArgs: someRelativeFolderName

[StatusLED]
enabled: True
Type: StatusLED
Pin: 10
```

**Comment by s-martin** [comment]
You have to make sure that every pin in gpio_settings.ini is defined only once. 

You can remove sections you don't use or at least comment them out with a leading `;`

**Comment by jonasschmid404** [comment]
ok, thank you. So while we're at it: I bought an external USB-Soundcard ([this one](https://www.amazon.de/dp/B0B45SKDG1?ref=ppx_yo2ov_dt_b_fed_asin_title)) and it does not seem to be configured properly. I can play out through the onboard headphone jack, but not through the usb sound card. I checked issue #128, but when I enter $ cat /proc/asound/modules with the soundcard attached, it gives me this output: 0 snd_bcm2835, 2 vc4. When initially installing, I went with the "PCM"-Standard.

**Comment by s-martin** [comment]
Not sure, if your soundcard is recognized.

please check here for audio troubleshooting: https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Troubleshooting-FAQ#audio-is-not-working
